Packages/vcs/vcs/vcsvtk/boxfillpipeline.py

```diff
@@ -286,7 +245,7 @@ def _plotInternalBoxfill(self):
 
         # Colortable bit
         # make sure length match
-        numLevels = len(self._contourLevels)
+        numLevels = len(self._contourLevels) - 1
```

Comment:
It was wrong before. When we use it right below, we pad the colors in `_contourColors` to the length of `numLevels`, but `_contourLevels` represents the bounds of each "level"; this means that there was one more color than there should have been, and led to a duplicated color appearing at the end of the legend (hard to see if you're using lots of levels, but very easy to see when you drop it down to a small number of levels). That's actually the root of this whole pull request, but I hit on some other sticky widgets along the way.